Live map crashing

Sine this morning we get complaints about freezing/crashing live map.
I can reproduce this after short usage of the live map.

As it wasnt the case last days I assume we have a server issue one again (or still) where cgeo crashes instead of catching the error and giving a proper error message.
Maybe someone can reproduce and provide a log.
IMHO we should urgently fix this because people blame cgeo for this.

@rsudev 
You mean e.g. regarding this reported crash there is nothing we can catch or work around?
```
java.lang.RuntimeException: java.net.UnknownHostException: Unable to resolve host "tiles.geocaching.com": No address associated with hostname
	at rx.exceptions.Exceptions.propagate(Exceptions.java:58)
	at rx.singles.BlockingSingle.value(BlockingSingle.java:93)
	at cgeo.geocaching.connector.gc.Tile.requestMapInfo(Tile.java:237)
	at cgeo.geocaching.connector.gc.GCMap.searchByGeocodes(GCMap.java:68)
	at cgeo.geocaching.maps.CachesOverlay$RequestDetailsThread.run(CachesOverlay.java:295)
Caused by: java.net.UnknownHostException: Unable to resolve host "tiles.geocaching.com": No address associated with hostname
	at java.net.InetAddress.lookupHostByName(InetAddress.java:479)
	at java.net.InetAddress.getAllByNameImpl(InetAddress.java:259)
	at java.net.InetAddress.getAllByName(InetAddress.java:221)
	at okhttp3.Dns$1.lookup(Dns.java:39)
	at okhttp3.internal.connection.RouteSelector.resetNextInetSocketAddress(RouteSelector.java:172)
	at okhttp3.internal.connection.RouteSelector.nextProxy(RouteSelector.java:138)
	at okhttp3.internal.connection.RouteSelector.next(RouteSelector.java:80)
	at okhttp3.internal.connection.StreamAllocation.findConnection(StreamAllocation.java:178)
	at okhttp3.internal.connection.StreamAllocation.findHealthyConnection(StreamAllocation.java:129)
	at okhttp3.internal.connection.StreamAllocation.newStream(StreamAllocation.java:98)
	at okhttp3.internal.connection.ConnectInterceptor.intercept(ConnectInterceptor.java:42)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:92)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:67)
	at okhttp3.internal.cache.CacheInterceptor.intercept(CacheInterceptor.java:109)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:92)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:67)
	at okhttp3.internal.http.BridgeInterceptor.intercept(BridgeInterceptor.java:93)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:92)
	at okhttp3.internal.http.RetryAndFollowUpInterceptor.intercept(RetryAndFollowUpInterceptor.java:124)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:92)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:67)
	at cgeo.geocaching.network.Network$HeadersInterceptor.intercept(Network.java:215)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:92)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:67)
	at okhttp3.RealCall.getResponseWithInterceptorChain(RealCall.java:170)
	at okhttp3.RealCall.access$100(RealCall.java:33)
	at okhttp3.RealCall$AsyncCall.execute(RealCall.java:120)
	at okhttp3.internal.NamedRunnable.run(NamedRunnable.java:32)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1112)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:587)
	at java.lang.Thread.run(Thread.java:818)
Caused by: android.system.GaiException: android_getaddrinfo failed: EAI_NODATA (No address associated with hostname)
	at libcore.io.Posix.android_getaddrinfo(Native Method)
	at libcore.io.ForwardingOs.android_getaddrinfo(ForwardingOs.java:55)
	at java.net.InetAddress.lookupHostByName(InetAddress.java:445)
	... 30 more
```

--- a/main/res/values/strings.xml
+++ b/main/res/values/strings.xml
@@ -254,6 +254,7 @@
     <string name="err_trackable_log_not_anonymous">Logging %1$s from c:geo cannot be done anonymously. Do you want to open the settings to fill in your credentials from %2$s?</string>
     <string name="err_trackable_no_preference_activity">Sorry, cannot find a suitable preference screen.</string>
     <string name="warn_discard_changes">Unsaved waypoint changes discarded!</string>
+    <string name="err_request_popup_info">c:geo failed to get cache popup info.</string>
 
     <!-- location service -->
     <string name="loc_last">Last known</string>

--- a/main/src/cgeo/geocaching/maps/CachesOverlay.java
+++ b/main/src/cgeo/geocaching/maps/CachesOverlay.java
@@ -3,6 +3,7 @@ package cgeo.geocaching.maps;
 import cgeo.geocaching.CachePopup;
 import cgeo.geocaching.R;
 import cgeo.geocaching.WaypointPopup;
+import cgeo.geocaching.activity.ActivityMixin;
 import cgeo.geocaching.activity.Progress;
 import cgeo.geocaching.connector.gc.GCMap;
 import cgeo.geocaching.enumerations.CacheType;
@@ -23,9 +24,7 @@ import cgeo.geocaching.settings.Settings;
 import cgeo.geocaching.storage.DataStore;
 import cgeo.geocaching.utils.Log;
 
-import org.apache.commons.lang3.StringUtils;
-import android.support.annotation.NonNull;
-
+import android.app.Activity;
 import android.content.Context;
 import android.content.res.Resources.NotFoundException;
 import android.graphics.Canvas;
@@ -35,10 +34,13 @@ import android.graphics.Paint.Style;
 import android.graphics.PaintFlagsDrawFilter;
 import android.graphics.Point;
 import android.location.Location;
+import android.support.annotation.NonNull;
 
 import java.util.ArrayList;
 import java.util.Collections;
 import java.util.List;
+
+import org.apache.commons.lang3.StringUtils;
 
 public class CachesOverlay extends AbstractItemizedOverlay {
 
@@ -292,7 +294,12 @@ public class CachesOverlay extends AbstractItemizedOverlay {
         @Override
         public void run() {
             if (requestRequired()) {
+                try {
                 /* final SearchResult search = */GCMap.searchByGeocodes(Collections.singleton(cache.getGeocode()));
+                } catch (final Exception ex) {
+                    Log.w("Error requesting cache popup info", ex);
+                    ActivityMixin.showToast((Activity) context, R.string.err_request_popup_info);
+                }
             }
             CGeoMap.markCacheAsDirty(cache.getGeocode());
             CachePopup.startActivity(context, cache.getGeocode());

--- a/main/src/cgeo/geocaching/maps/mapsforge/v6/NewMap.java
+++ b/main/src/cgeo/geocaching/maps/mapsforge/v6/NewMap.java
@@ -1189,7 +1189,7 @@ public class NewMap extends AbstractActionBarActivity {
                 }
             };
 
-            AlertDialog dialog = new AlertDialog.Builder(this)
+            final AlertDialog dialog = new AlertDialog.Builder(this)
                 .setTitle(res.getString(R.string.map_select_multiple_items))
                 .setAdapter(adapter, new SelectionClickListener(sorted))
                 .create();
@@ -1290,7 +1290,12 @@ public class NewMap extends AbstractActionBarActivity {
                 return;
             }
             if (requestRequired()) {
+                try {
                 /* final SearchResult search = */GCMap.searchByGeocodes(Collections.singleton(cache.getGeocode()));
+                } catch (final Exception ex) {
+                    Log.w("Error requesting cache popup info", ex);
+                    ActivityMixin.showToast(map, R.string.err_request_popup_info);
+                }
             }
             map.popupGeocodes.add(cache.getGeocode());
             CachePopup.startActivityAllowTarget(map, cache.getGeocode());

